src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -90,253 +139,3541 @@ static ColliderJntSphInit D_80A65F28 = {
+            } else {
+                EnHorse_ObstructMovement(this, globalCtx, 5, galloping);
+            }
+            return;
+        }
+
+        if (this->flags & ENHORSE_JUMPING) {
+            this->actor.shape.rot.x = 0;
+            return;
+        }
+
+        if (this->actor.floorHeight + 4.0f < this->actor.world.pos.y) {
+            this->actor.shape.rot.x = 0;
+            return;
+        }
+
+        if (fabsf(floorSlope) > 8191.0f) {
+            return;
+        }
+
+        this->actor.shape.rot.x = floorSlope;
+        this->actor.shape.yOffset =
+            (this->yFront + (((this->yBack - this->yFront) * 20.0f) / 45.0f)) - this->actor.floorHeight;
+    }
+}
+
+s32 EnHorse_GetMountSide(EnHorse* this, GlobalContext* globalCtx);
+void EnHorse_MountDismount(EnHorse* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s32 mountSide;
+    Player* player = PLAYER;
+
+    mountSide = EnHorse_GetMountSide(this, globalCtx);
+    if (mountSide != 0 && !(this->flags & ENHORSE_UNRIDEABLE) && player->rideActor == NULL) {
+        Actor_SetRideActor(globalCtx, &this->actor, mountSide);
+    }
+
+    if (this->playerControlled == 0 && Actor_IsMounted(globalCtx, &this->actor) == 1) {
+        this->noInputTimer = 55;
+        this->noInputTimerMax = 55;
+        this->playerControlled = 1;
+        EnHorse_Freeze(this);
+    } else if (this->playerControlled == 1 && Actor_NotMounted(globalCtx, &this->actor) == 1) {
+        this->noInputTimer = 35;
+        this->noInputTimerMax = 35;
+        this->flags &= ~ENHORSE_UNRIDEABLE;
+        this->playerControlled = 0;
+        EnHorse_Freeze(this);
+    }
+}
+
+void EnHorse_StickDirection(Vec2f* curStick, f32* stickMag, s16* angle) {
+    f32 dist;
+    f32 y;
+    f32 x;
+
+    x = curStick->x;
+    y = curStick->y;
+    dist = sqrtf(SQ(x) + SQ(y));
+
+    *stickMag = dist;
+    if (dist > 60.0f) {
+        *stickMag = 60.0f;
+    } else {
+        *stickMag = *stickMag;
+    }
+
+    *angle = Math_FAtan2F(-curStick->x, curStick->y) * (32768.0f / M_PI);
+}
+
+void EnHorse_UpdateStick(EnHorse* this, GlobalContext* globalCtx) {
+    this->lastStick = this->curStick;
+    this->curStick.x = globalCtx->state.input[0].rel.stick_x;
+    this->curStick.y = globalCtx->state.input[0].rel.stick_y;
+}
+
+void EnHorse_ResolveCollision(EnHorse* this, GlobalContext* globalCtx, CollisionPoly* colPoly) {
+    f32 dist;
+    f32 nx;
+    f32 ny;
+    f32 nz;
+    f32 offset;
+
+    nx = colPoly->normal.x * 0.00003051851f;
+    ny = colPoly->normal.y * 0.00003051851f;
+    nz = colPoly->normal.z * 0.00003051851f;
+    if (Math_CosS(this->actor.world.rot.y - (s16)(Math_FAtan2F(colPoly->normal.x, colPoly->normal.z) * 10430.378f) - 0x7FFF) <
+        0.7071f) {
+        return;
+    }
+
+    dist = Math3D_DistPlaneToPos(nx, ny, nz, colPoly->dist, &this->actor.world.pos);
+    offset = (1.0f / sqrtf(SQ(nx) + SQ(nz)));
+    offset = (30.0f - dist) * offset;
```

Comment:
may want to negate condition and have the block of code inside? not sure which reads nicer

Reply:
I needed to use `!(...)` since it's floating point